lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -368,6 +365,18 @@ async sub transform_transaction {
             timestamp    => $int_timestamp,
         );
 
+        # if the gas is empty we don't proceed
+        unless ($gas && $decoded_transaction->{gasPrice}) {
+            my $our_transaction = await $self->_set_transaction_type($transaction);
```

Comment:
at this point it doesn't seem that we have `$self->accounts` defined yet, so this will always return undef, seems in #L384 we populate accounts object.

Reply:
Thanks, updated.